### PR TITLE
docs: Migrate landing page content and hugo scaffold into the Go CDK repo

### PR DESCRIPTION
**Pull request by eliben**

### CODE REVIEW

File: internal/website/README.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+Source for the TODO website. Powered by [Hugo](https://gohugo.io/).
```
Comment (ijt) [inline internal/website/README.md]: Why TODO? Don't we know it's gocloud.dev?

File: internal/website/themes/hyde/CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# Changelog
```
Comment (ijt) [inline internal/website/themes/hyde/CHANGELOG.md]: It's annoying to have all this theme stuff in here. Isn't there something we can do to only have the real content?

File: internal/website/README.md (hunk 1 above)
Comment (eliben) [reply internal/website/README.md]: Most likely, yes. I figured we can fill this in later when we're ready to push it out.

File: internal/website/themes/hyde/CHANGELOG.md (hunk 2 above)
Comment (eliben) [reply internal/website/themes/hyde/CHANGELOG.md]: I'm not sure what. We could extract only the essential parts of the theme - that would reduce the line count but would require significant time investment. The theme was modified for our needs so it's not something we can just link to outside, and it's essential for Hugo to generate the HTML.

File: internal/website/layouts/index.html [hunk 3]
```diff
@@ -0,0 +1,4 @@
+{{ define "main" -}}
+{{ .Content }}
+</div>
```
Comment (zombiezen) [inline internal/website/layouts/index.html]: Mismatched div?

Comment (eliben) [reply internal/website/layouts/index.html]: Nice spot! Fixed

**Pull request merged**